Tests/SonarScanner.MSBuild.PreProcessor.Test/PreProcessorTests.cs

```diff
@@ -60,111 +55,64 @@ public async Task PreProc_EndToEnd_SuccessCase()
             // * server properties are fetched
             // * rule sets are generated
             // * config file is created
+            using var scope = new TestScope(TestContext);
+            var factory = new MockObjectFactory();
+            var settings = factory.ReadSettings();
```

Comment:
There are 3 commits:
Phase 1: Rework `MockObjectFactory` - complex change
Phase 2: Extract private class `TestScope` to avoid duplications - smaller
Phase 3: Extract `ReadSettings` to avoid duplications - smaller commit